src/coreclr/jit/importer.cpp

```diff
@@ -4573,6 +4614,120 @@ GenTree* Compiler::impTypeIsAssignable(GenTree* typeTo, GenTree* typeFrom)
     return nullptr;
 }
 
+GenTree* Compiler::impUnrollSpanComparisonAgainstConst(GenTree* span, GenTree* constSpan, bool ignoreCase, bool startsWith)
+{
+#ifdef TARGET_64BIT
+    // TODO: Implement ignoreCase support.
+    assert(!ignoreCase);
+
+    if (!constSpan->OperIs(GT_CNS_STR))
+    {
+        // For now we only support constant strings
+        return nullptr;
+    }
+
+    GenTreeStrCon* cnsStr = constSpan->AsStrCon();
+
+    // Grab the actual string literal from VM
+    int       strLen = -1;
+    LPCWSTR   str    = info.compCompHnd->getStringLiteral(cnsStr->gtScpHnd, cnsStr->gtSconCPX, &strLen);
+    var_types cmpType;
+
+    if (strLen == 0)
+    {
+        if (startsWith)
+        {
+            return gtNewIconNode(1);
+        }
+        else
+        {
+            // TODO: Emit `span.Length == 0`
+            return nullptr;
+        }
+    }
+    if (strLen == 1)
+    {
+        cmpType = TYP_BYTE;
```

Comment:
It's UTF-16 strings, so should this be `TYP_SHORT` instead of byte?
Or is it OK as belowe there's a check for ASCII?

Reply:
Oops, indeed, thanks! I'm going to add tests for such cases.